provisioning/baremetal_config.go

```diff
@@ -0,0 +1,116 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provisioning
+
+import (
+	"fmt"
+
+	metal3iov1alpha1 "github.com/openshift/cluster-baremetal-operator/api/v1alpha1"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+const (
+	// BaremetalProvisioningCR is the name of the provisioning resource
+	BaremetalProvisioningCR = "provisioning-configuration"
+)
+
+var (
+	log = ctrl.Log.WithName("provisioning").WithName(BaremetalProvisioningCR)
+)
+
+// ValidateBaremetalProvisioningConfig validates the contents of the provisioning resource
+func ValidateBaremetalProvisioningConfig(prov *metal3iov1alpha1.Provisioning) error {
+	provisioningNetworkMode := getProvisioningNetworkMode(prov)
+	log.V(1).Info("Final Provisioning Network", "Mode", provisioningNetworkMode)
+	switch provisioningNetworkMode {
```

Comment:
We don't specify an enum, yet, so this value isn't guaranteed to be one of the 3 valid values. We'll need a default handler here to report an error, in case someone uses an invalid value.

Reply:
Added validation for Managed as the default case.

Reply:
After making ProvisioningNetwork and enum in the CRD, we can go back to not having a default case.